person-directory-impl/src/main/java/org/apereo/services/persondir/support/ScriptEnginePersonAttributeDao.java

```diff
@@ -158,6 +174,9 @@ public IPersonAttributes getPerson(final String uid, final IPersonAttributeDaoFi
                                                                      final IPersonAttributeDaoFilter filter) {
         final Set<IPersonAttributes> people = new LinkedHashSet<>();
         final String username = usernameAttributeProvider.getUsernameFromQuery(query);
+        if (username == null) {
+            logger.trace("Unable to find username in map {} using attribute {}", query, usernameAttributeProvider.getUsernameAttribute());
+        }
```

Comment:
Should you return null here?

Reply:
I wanted to log trace there since there is more information available but if I return there then I would want to raise the log level to WARN. That has the disadvantage that if the script dao is disabled (not sure if that is configurable) then the WARN message would show up. I figured the other method should check for null anyway and that it can do the WARN message if the script is enabled.